### PR TITLE
[vmware] `az vmware cluster create`: Fix the mandatory flag of `--hosts`

### PR DESCRIPTION
**Pull request by jonathanhe-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by jonathanhe-msft** [comment]
@cataggar please review

File: src/vmware/azext_vmware/tests/latest/test_vmware_scenario.py [hunk 1]
```diff
@@ -82,6 +82,9 @@ def test_vmware(self):
         # cluster create
         self.cmd('vmware cluster create -g {rg} -c {privatecloud} -n {cluster} --sku av20 --size 3 --hosts {hosts}')
 
+        # cluster create without --hosts
+        self.cmd('vmware cluster create -g {rg} -c {privatecloud} -n {cluster} --sku av20 --size 3')
```
Comment (cataggar) [inline src/vmware/azext_vmware/tests/latest/test_vmware_scenario.py:86]: Thanks for the unit test.

**Review by cataggar — APPROVED** [review]
PR content looks good. PR title could use a better title.

File: src/vmware/HISTORY.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Release History
 
+## 5.0.3 (2023-06)
+
+- Bug fix for `az vmware cluster create` [#26653](https://github.com/Azure/azure-cli/issues/26653)
```
Comment (zhoxing-ms) [inline src/vmware/HISTORY.md:5]:
```suggestion
- Fix the mandatory flag of `--hosts` for `az vmware cluster create` [#26653](https://github.com/Azure/azure-cli/issues/26653)
```

**Comment by jonathanhe-msft** [comment]
@zhoxing-ms thanks for the review. I updated based on your comments, please take another look. thanks!

File: src/vmware/HISTORY.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Release History
 
+## 5.0.3 (2023-06)
```
Comment (zhoxing-ms) [inline src/vmware/HISTORY.md:3]:
```suggestion
## 5.0.3 (2023-07)
```

**Comment by jonathanhe-msft** [comment]
@zhoxing-ms  can you please take another look at this?

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**